nqeit-cross-engine/src/test/java/org/projectnessie/integtests/crossengine/ITCrossEngineExternalNessie.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.integtests.crossengine;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.spark.sql.SparkSession;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.projectnessie.integtests.dremio.DremioHelper;
+import org.projectnessie.integtests.dremio.IcebergDremioExtension;
+import org.projectnessie.integtests.flink.Flink;
+import org.projectnessie.integtests.flink.FlinkHelper;
+import org.projectnessie.integtests.flink.IcebergFlinkExtension;
+import org.projectnessie.integtests.iceberg.spark.IcebergSparkExtension;
+import org.projectnessie.integtests.iceberg.spark.Spark;
+import org.projectnessie.integtests.nessie.NessieTestsExtension;
+
+@ExtendWith({
+  IcebergSparkExtension.class,
+  IcebergFlinkExtension.class,
+  NessieTestsExtension.class,
+  IcebergDremioExtension.class
+})
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class ITCrossEngineExternalNessie {
+
+  private static SparkSession spark;
+  private static FlinkHelper flink;
+  private static DremioHelper dremioHelper;
+
+  @BeforeAll
+  public static void maybeSkipTests() {
```

Comment:
Nothing guarantees that this method's called before `setupEngines`

Reply:
The problem of using multiple @BeforeAll is been solved using interface. Added an interface class named Dremio. 

Doc states that "@BeforeAll methods from an interface will be executed before @BeforeAll methods in the class that implements the interface"
Reference used: https://junit.org/junit5/docs/5.4.2/api/org/junit/jupiter/api/BeforeAll.html

Reply:
(From above) The new extension could easily do that skip, which is much cleaner than a @Before(All) or an interface.

Reply:
Used extension to skip the tests if the `dremio-url` is not provided, and fail hard if any of the parameter is missing.